pkg/js/helpers.js

```diff
@@ -304,6 +313,33 @@ function addRecord(d,type,name,target,mods) {
     return rec;
 }
 
+function addRecordSRV(d,type,name,priority,weight,port,target,mods) {
+    var rec = {type: type, name: name, priority: priority, weight: weight, port: port, target: target, ttl:d.defaultTTL, meta:{}};
+    // for each modifier, decide based on type:
+    // - Function: call is with the record as the argument
+    // - Object: merge it into the metadata
+    // FIXME(tlim): Factor this code out to its own function.
```

Comment:
instead of having a separate `addRecordSRV` function, just do `var rec = addRecord(d, "SRV", name, target, mods)` then set weight / port and those fields on the returned record.